x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -278,7 +278,7 @@ public void authorizeIndexAction(RequestInfo requestInfo, AuthorizationInfo auth
                     // the same as the user that submitted the original request so we can skip security here.
                     listener.onResponse(new IndexAuthorizationResult(true, IndicesAccessControl.ALLOW_NO_INDICES));
                 }
-            } else if (action.equals(ClearReaderAction.NAME)) {
+            } else if (isReaderContextsRelatedAction(action)) {
```

Comment:
I don't think we should do this for `ShardOpenReaderRequest`. Can we implement IndicesRequest instead so that opening a reader also checks the index permissions ?

Reply:
Good catch

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/55265/commits/791008740df068447f5269ff05a126216eaad70b